### DISCUSSION TITLE
Having trouble installing xrspatial for zonal statistics

### DISCUSSION THREAD

**Discussion by advythr** [earthlab-education/Earth-Analytics-2023-01-Intro#679] [! GitHub help] [answered]
Hi everyone,

For my final project I am trying to calculate zonal means for polygons from a raster layer that is an xarray DataArray. To do this I am trying to find a package that calculates zonal statistics.

When I install xarray-spatial using pip install the install seems to be successful.
<img width="1434" alt="Screen Shot 2023-12-15 at 9 57 13 AM" src="https://github.com/earthlab-education/Earth-Analytics-2023-01-Intro/assets/83184164/31d29363-9592-416f-860a-8628e4e00d6a">

But when I use this code to import the zonal_stats function,

```
from xrspatial import zonal_stats
```

I get this error:
<img width="1452" alt="Screen Shot 2023-12-15 at 9 58 25 AM" src="https://github.com/earthlab-education/Earth-Analytics-2023-01-Intro/assets/83184164/7eaaa532-9d5a-43b3-93ec-778a68f4a78c">

Anyone have an idea on how to fix this?

I am running this code in VS Code with Jupyter Lab on a M1 Macbook Pro, using the Earth Analytics Python environment.

**Comment by eculler** [comment, marked as answer]
Looks like you installed in your base environment, not your earth-analytics-python environment. Make sure to activate before the install

**Reply by advythr** [reply]
Hi Elsa,

Thanks! I successfully installed it using:

```bash
conda activate earth-analytics-python
```

and then
```
pip install xarray-spatial
```

and then this code worked:
```
from xrspatial import zonal stats
```
